spec/consensus/gossip/crdt.md

```diff
@@ -130,48 +129,178 @@ Consider the following definition for **Eventual Convergence**.
+Instead they assume that replicas will compare their states and converge two-by-two using a merge function;
+as long as the function is commutative, associative and idempotent, the states will converge.
+For example, in the G-Set case, the merge operator is simply the union of the sets.
+
+The two approaches for converging the message sets in the Tendermint algorithm described [earlier](#nodes-state-as-a-tuple-space), without the deletion of entries, correspond to the operation- and state-based [Grow-only Set](https://en.wikipedia.org/wiki/Conflict-free_replicated_data_type#G-Set_(Grow-only_Set)) CRDT;
+if removals must be handled, then using a 2P-Set is an option.
+To the best of our knowledge, no existing CRDT supports superseding of elements, until now.
+
+### About supersession
+
+We call an entry of the tuple space an **Entry**[^tuple] and call **View** a set of Entry.
+
+[^tuple]: We use Entry instead of Tuple not to conflict with the use of tuple in Quint.
+
+Let $v_1, v_2, v_3$ be **View**.
+We say that $v_1$ is superseded by $v_2$ if all elements of $v_1$ are rendered stale by some subset of $v_2$ and note this relation as $v_1 < v_2$.
+For all $e \in v_1$, if $v_1 < v_2$ then we abuse the notation and terminology and simply say that $e$ is superseded by $v_2$ and note it as $e < v_2$
+
+The supersession $<$ relation must respect the following properties:[^not_contains]
+
+[^not_contains]: $v_1 \subset v_2 \implies v_1 < v_2$ but $v_1 < v_2 \not\implies v_1 \subset v_2$
+
+1. Transitivity: if $v_1 < v_2$ and $v_2 < v_3$, then $v_1 < v_3$;
+1. Reflexiveness: $v < v$;
+1. Anti-symmetry: if $v_1 < v_2$ and $v_3 < v_2$ then $v_1 = v_2$;
+1. Tombstone validity: if $v_1 < v_2$, then $v_1$ is superseded by any sets obtained by replacing entries in $v_2$ by their corresponding tombstones;[^tomb_ss]
+
+[^tomb_ss]: In other words, $e < \{\bar{e}\}$
+
+### Superseding Views CRDT
+
+We define here the Superseding Views CRDT (SSE)[^name] as a set in which superseded elements are removed as new elements are added.
+Since superseded elements are irrelevant from the point of view of CRDT users, they may be removed without prejudice as long as the CRDT properties are ensured.
+
+[^name]: settle on a name.
+
+We define SSE is as a state-based CRDT. 
+While an equivalent operations-based definition must exist, it is out of the scope of this document.
+
+
+#### The merge operator - $\sqcup$
+
+State based CRDT use a merge operation to combine two replica states.
+In our CRDT we embed the removal of superseded messages, as defined in the previous section, in the merge operation.
+
+Let $\mathcal{V}$ be the power set of entries, or the set of all views of a system.
+The merge operator $\sqcup: \mathcal{V} x \mathcal{V} \rightarrow \mathcal{V}$ is defined as the union of the input views, without the entries superseded by the other elements in the union, that is: $v_1 \sqcup v_2 = [e: e \in (v_1 \cup v_2) \land e < (v_1 \cup v_2 \setminus \{e\})]$
```

Comment:
Unclear that this is actually returning the least upper bound of $\\{v_1,v_2\\}$. 

If the superseding relation is defined as proposed in the comment above, we may construct a merge operator as follows: starting from v_2, one iterates over all the elements of v_1. For each of them e, if e is superseding it is added and all the lower elements removed from the resulting set. (There might be a better implem. thought. For that, we can look at soa CRDT implementations.)